src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -0,0 +1,237 @@
+                    return VisitExpressionWithStackGuard(ref _recursionDepth, expr);
+                }
+                return base.Visit(node);
+            }
+
+            private bool _disposedValue = false;
+
+            void Dispose(bool disposing)
+            {
+                if (!_disposedValue)
+                {
+                    if (disposing)
+                    {
+                        _topLevelNullabilityMap.Free();
+                    }
+
+                    _disposedValue = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+            }
+
+            public override BoundNode VisitAnonymousObjectCreationExpression(BoundAnonymousObjectCreationExpression node)
+            {
+
+                if (node.Declarations.Length != node.Arguments.Length)
+                {
+                    // https://github.com/dotnet/roslyn/issues/24018 implicit anonymous properties are not currently handled,
+                    // so skip verification on the declarations for now and only verify the AnonymousObjectCreation and the
+                    // arguments.
+                    foreach (var arg in node.Arguments)
+                    {
+                        Visit(arg);
+                    }
+                    return null;
+                }
+                else
+                {
+                    return base.VisitAnonymousObjectCreationExpression(node);
+                }
+            }
+
+            public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationExpression node)
+            {
+                // If the delegate creation was resolved to a single static method, we do not examine the
+                // receiver of the method group as it's completely unused.
+                if (node.Argument is BoundMethodGroup group)
+                {
+                    VerifyNode(group);
+                    if (node.MethodOpt?.IsStatic == false)
+                    {
+                        Visit(group.ReceiverOpt);
+                    }
+
+                    return null;
+                }
+                else
+                {
+                    return base.VisitDelegateCreationExpression(node);
+                }
+            }
+
+            public override BoundNode VisitNewT(BoundNewT node)
+            {
+                // https://github.com/dotnet/roslyn/issues/33387 We're not currently
+                // examining child nodes correctly
+                if (node.InitializerExpressionOpt != null)
+                {
+                    VerifyNode(node.InitializerExpressionOpt, overrideSkippedExpression: true);
+                }
+
+                return null;
+            }
+
+            public override BoundNode VisitBadExpression(BoundBadExpression node)
+            {
+                // Regardless of what the BadExpression is, we need to add all of its direct children to the visited map. They
```

Comment:
>we need to add all of its direct children to the visited map [](start = 60, length = 60)

That seems unnecessary. We should consider having the consumer of the map gracefully handle missing entries.

Reply:
Well, with the changes that Neal is making with regards to invalid expressions that might be ok, but at least when I started this PR that would not be OK because we want to make sure that we're actually visiting things. Regardless, this particular exception isn't about missing entries, it's about things with extra entries.

---
In reply to: [260070678](https://github.com/dotnet/roslyn/pull/33591#discussion_r260070678) [](ancestors = 260070678)

Reply:
Why add those extra entries though since we don't report warnings within those entries in `NullableWalker`?

---
In reply to: [261437377](https://github.com/dotnet/roslyn/pull/33591#discussion_r261437377) [](ancestors = 261437377,260070678)